contrib/kind.sh

```diff
@@ -205,307 +203,339 @@ print_params()
-  if [ "$KIND_REMOVE_TAINT" == true ]; then
-    # do not error if it fails to remove the taint
-    kubectl taint node $n node-role.kubernetes.io/master:NoSchedule- || true
+kubectl_apply_ingress() {
+  # Delete kube-proxy
+  run_kubectl -n kube-system delete ds kube-proxy
+  kind get clusters
+  kind get nodes --name ${KIND_CLUSTER_NAME}
+  kind export kubeconfig --name ${KIND_CLUSTER_NAME}
+  if [ "$KIND_INSTALL_INGRESS" == true ]; then
+    run_kubectl apply -f ingress/mandatory.yaml
+    run_kubectl apply -f ingress/service-nodeport.yaml
   fi
-done
-if [ "$OVN_HA" == true ]; then
-  run_kubectl apply -f ovnkube-db-raft.yaml
-else
-  run_kubectl apply -f ovnkube-db.yaml
-fi
-run_kubectl apply -f ovs-node.yaml
-run_kubectl apply -f ovnkube-master.yaml
-run_kubectl apply -f ovnkube-node.yaml
-popd
+}
 
-# Delete kube-proxy
-run_kubectl -n kube-system delete ds kube-proxy
-kind get clusters
-kind get nodes --name ${KIND_CLUSTER_NAME}
-kind export kubeconfig --name ${KIND_CLUSTER_NAME}
-if [ "$KIND_INSTALL_INGRESS" == true ]; then
-  run_kubectl apply -f ingress/mandatory.yaml
-  run_kubectl apply -f ingress/service-nodeport.yaml
-fi
+kubectl_wait_pods() {
+  # Check that everything is fine and running. IPv6 cluster seems to take a little
+  # longer to come up, so extend the wait time.
+  OVN_TIMEOUT=300s
+  if [ "$KIND_IPV6_SUPPORT" == true ]; then
+    OVN_TIMEOUT=480s
+  fi
+  if ! kubectl wait -n ovn-kubernetes --for=condition=ready pods --all --timeout=${OVN_TIMEOUT} ; then
+    echo "some pods in OVN Kubernetes are not running"
+    kubectl get pods -A -o wide || true
+    exit 1
+  fi
+  if ! kubectl wait -n kube-system --for=condition=ready pods --all --timeout=300s ; then
+    echo "some pods in the system are not running"
+    kubectl get pods -A -o wide || true
+    exit 1
+  fi
 
-# Check that everything is fine and running. IPv6 cluster seems to take a little
-# longer to come up, so extend the wait time.
-OVN_TIMEOUT=300s
-if [ "$KIND_IPV6_SUPPORT" == true ]; then
-  OVN_TIMEOUT=480s
-fi
-if ! kubectl wait -n ovn-kubernetes --for=condition=ready pods --all --timeout=${OVN_TIMEOUT} ; then
-  echo "some pods in OVN Kubernetes are not running"
-  kubectl get pods -A -o wide || true
-  exit 1
-fi
-if ! kubectl wait -n kube-system --for=condition=ready pods --all --timeout=300s ; then
-  echo "some pods in the system are not running"
-  kubectl get pods -A -o wide || true
-  exit 1
-fi
+  echo "Pods are all up, allowing things settle for 30 seconds..."
+  sleep 30
+}
 
-echo "Pods are all up, allowing things settle for 30 seconds..."
-sleep 30
+# In order to allow providing arguments with spaces, e.g. "-vconsole:info -vfile:info" 
+# the original command <parse_args $*> was replaced by <parse_args "$@">
+parse_args "$@"
+install_j2_renderer
+set_params
+print_params
+set -euxo pipefail
+detect_apiserver_ip
+check_ipv6
+set_cidr_params
+create_kind_cluster
+docker_disable_ipv6
+coredns_patch_for_github_ci
```

Comment:
let's move the condition to the main loop so we can see the flow of the script in one place and we don't have to check all the functions
```
  if [ "${GITHUB_ACTIONS:-false}" == "true" ]; then
      coredns_patch_for_github_ci
  fi
```

Reply:
ok